File: benchmark/c/cpw/number_is_divisible_by_29_or_not/number_is_divisible_by_29_or_not.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int number_is_divisible_by_29_or_not ( long int n ) ;
int number_is_divisible_by_29_or_not ( long int n ) {
  while ( n / 100 ) {
    int last_digit = n % 10;
    n /= 10;
    n += last_digit * 3;
  }
  return ( n % 29 == 0 );
}

File: translation_result/else_bms/fail/c/cpw/number_is_divisible_by_29_or_not/gemini/Restart/number_is_divisible_by_29_or_not.rs


fn number_is_divisible_by_29_or_not(n: i64) -> bool {
    let mut num = n;
    while num / 100 > 0 {
        let last_digit = (num % 10) as i64;
        num /= 10;
        num += last_digit * 3;
    }
    (num % 29) == 0
}
